modules/ocf/files/packages/cups/raster-filter

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# A Tea4CUPS filter to perform rasterization.
+# Based off the pdf-open script
+
+set -euo pipefail
+
+input=$(mktemp --suffix='.pdf')
+
+# Feed filter input into temporary file
+cat - > "$input"
+
+# If the file is not a PDF, then just pass it thorugh
+# (Note that all PDFs start with "%PDF")
+if [ "$(head -c 4 "$input")" != "%PDF" ]; then
+    cat "$input"
+    exit
+fi
+
+# Rasterizing PDF...
+output=$(mktemp --suffix='.pdf')
+
+if ! convert -density 250 "$input" "$output" 2> /tmp/error; then
```

Comment:
Instead of using a tempfile here, you can replace `"$output"` with `-` to output directly to stdout.

Reply:
Is /tmp/error ever checked? What's the purpose of this file?

Reply:
Alas, `/tmp/error` isn't checked. I think I just wanted any error messages to get redirected somewhere else for debugging, and kind of left that there.

@dkess made a comment about having the script send out an email upon `convert` failure, which I'll look into implementing. Also, I'll remove the `output` temporary file and simply output to stdout.

Reply:
Actually, it seems like `convert` writes its result (to the specified output file) as it runs, so in the case that `convert` fails, the result will be a mangled PDF. Thus, perhaps outputting directly to stdout may not be a good idea, since we need to first check that `convert` successfully ran before sending off the result to printhost. In that case, I'll keep the temporary output file.

Reply:
Hmm, in that case, sure, you can use temporary files.